MSAL/src/MSALError_Internal.m

```diff
@@ -120,6 +120,19 @@ void MSALLogError(id<MSALRequestContext> ctx, NSString *domain, NSInteger code,
     return [NSError errorWithDomain:domain code:code userInfo:[NSDictionary dictionaryWithDictionary:userInfo]];
 }
 
+NSError *MSALCreateErrorWithUserInfo(NSString *domain, NSInteger code, NSString *errorDescription, NSString *oauthError, NSString *subError, NSError* underlyingError, NSDictionary* additionalUserInfo)
```

Comment:
nit: we can replace MSALCreateError with a call to this method with nil as an underlying error perhaps?
I am also fine with just adding additional user info as an extra parameter to the MSALCreateError.

Reply:
just wanted to avoid passing one more nil everywhere :) But I agree that we should keep the number of options low, updated.